### PR TITLE
Comment details pending moderation redesign

### PR DESCRIPTION
**Pull request by jarvislin**
See https://github.com/orgs/Automattic/projects/944/views/1?pane=issue&itemId=60397816
Design yWt5gg3nWORhu079Qfv3mS-fi-1135_4600

This PR mainly implemented the redesign of pending moderation, but also removed lot of logic related to unused UI component.

| Screenshot 1 | Screenshot 2 |
|--|--|
| Screenshot_20240528-161610 | Screenshot_20240528-161624 |

-----

## To Test:

1. Sign in JP app
2. Go to `Notifications` tab
3. Click on a notification with comment type
4. It should display the UI of pending status, currently the `Approve comment` button is not actionable
5. Click on `More options`
6. It should display a bottom sheet of moderation, currently the buttons on the bottom sheet are not actionable
7. Go to My site -> More -> Comments
8. Click on a comment
9. Repeat step 4 - 6.
10. Done, thank you!

-----

## Regression Notes

1. Potential unintended areas of impact

    - notification, comment

11. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

12. What automated tests I added (or what prevented me from doing so)

    - n/a

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java [hunk 1]
```diff
@@ -164,7 +146,6 @@ public abstract class CommentDetailFragment extends ViewPagerFragment implements
     @NonNull private EnumSet<EnabledActions> mEnabledActions = EnumSet.allOf(EnabledActions.class);
 
     @Nullable protected CommentDetailFragmentBinding mBinding = null;
-    @Nullable protected ReaderIncludeCommentBoxBinding mReplyBinding = null;
```
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java:167]: This time I removed logic related to `mReplyBinding`.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/comments/SharedCommentDetailFragment.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            CommentStatus.SPAM -> {}
+            CommentStatus.TRASH -> {}
+            CommentStatus.DELETED -> {}
+            CommentStatus.ALL -> {}
+            CommentStatus.UNREPLIED -> {}
+            CommentStatus.UNSPAM -> {}
+            CommentStatus.UNTRASH -> {}
+        }
+
+        mBinding?.layoutCommentPending?.handlePendingView() // todo: remove this line after PR review
```
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/SharedCommentDetailFragment.kt:39]: The UI of pending status is forced to display by this line, this is only for PR review usage.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java (hunk 1 above)
Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java:167]: Thank you for cleaning up the code Jarvis 🙇

**Review by antonis — APPROVED** [review]
Awesome work @jarvislin 🙇 
I've tested the implementation on a Pixel 8 Pro with Android 14 and a tablet simulator with Android 14. The code looks great and improves the codebase 🏅  
I've tested various configurations including RTL languages and everything worked as expected 🎉 
I've only noticed an issue with the bottom sheet starting in collapsed state. I've replicated this both on a phone and tablet.

<details>
<summary>Recording</summary>

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/24ddc187-da0a-4000-99bf-6c90d56897a8
</details>
<details>
<summary>Screenshot</summary>

Screenshot_20240528_184722
</details>

Feel free to merge this PR and we can handle this separately.

**Comment by jarvislin** [comment]
@antonis thank you for reviewing this PR and catching the UI issue.
The bug you mentioned is related to `peekHeight`, I've fixed it in [1cdf268](https://github.com/wordpress-mobile/WordPress-Android/pull/20894/commits/1cdf268a6d191793982fa9ee0ada89900b4c257d).

**Review by antonis — APPROVED** [review]
Thank you for the fix @jarvislin 🙇 
The code change looks good and the bottom sheet worked as expected in my test 🎉 

<details>
<summary>Screenshot</summary>

Screenshot_20240529_104603

</details>

**Pull request merged**